### PR TITLE
Fix findbugs warnings

### PR DESCRIPTION
**Pull request by pyssling**
This should hopefully fix all findbugs warnings, though maybe my version of findbugs doesn't match the one in jenkins, in which case we'll need to fix some more based on that output.

Most bug fixes are real bug fixes, but some are just masking the issue in a not very meaningful way, in these cases a larger redesign may be needed.

### CODE REVIEW

File: src/main/java/org/jenkinsci/plugins/valgrind/ValgrindBuilder.java [hunk 1]
```diff
@@ -119,7 +119,7 @@ public boolean perform(AbstractBuild build, Launcher launcher, BuildListener lis
 				deleteOldReports(build, listener);
 			
 			ValgrindExecutable valgrindExecutable = new ValgrindExecutable(launcher,
-					build.getEnvironment(null).expand(this.valgrindExecutable));
+					build.getEnvironment(listener).expand(this.valgrindExecutable));
```
Comment (pyssling) [inline src/main/java/org/jenkinsci/plugins/valgrind/ValgrindBuilder.java]: Assuming this is correct, this is what other plugins do.

File: src/main/java/org/jenkinsci/plugins/valgrind/ValgrindBuilder.java [hunk 2]
```diff
@@ -131,6 +131,10 @@ public boolean perform(AbstractBuild build, Launcher launcher, BuildListener lis
 					return false;
 			}
 		}
+                catch (RuntimeException e)
```
Comment (pyssling) [inline src/main/java/org/jenkinsci/plugins/valgrind/ValgrindBuilder.java]: Pure findbugs suggestion, may or may not be relevant.

File: src/main/java/org/jenkinsci/plugins/valgrind/model/ValgrindError.java [hunk 3]
```diff
@@ -37,10 +37,14 @@
 
 	public String toString()
 	{
-		return 
-		"kind: " + kind + "\n" +
-		"text: " + description + "\n" +
-		"stack: " + stacktrace.toString();
+                if (kind != null && description != null && stacktrace != null)
```
Comment (pyssling) [inline src/main/java/org/jenkinsci/plugins/valgrind/model/ValgrindError.java]: this could be done in a better way, but this is at least as safe as before.

**Comment by pyssling** [comment]
@eXistence  any possibility of having a look at this commit? Much appreciated. :-)

**Comment by eXistence** [comment]
thx for contributing :)

**Pull request closed without merging**